### PR TITLE
Add missing clinic and calendar tests for vaccine flow, fix various bugs in flow

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This PR
- Adds unit tests for the calendar page
- Adds unit tests for the clinics page, including helpers for setting the facility and clinic for other tests
- Fixes bug with validation messages not showing on calendar, as well as fixed spacing when they do display
- Adds unit tests to cover maintenance warnings and facility check errors
- Fixed focus handling on facility availability check
- Refactored urls to use match.url in some top level vaccine flow components, to make testing easier
- Stopped direct booking criteria api from being called twice unnecessarily

## Testing done
Local and unit testing

## Screenshots
Fixed calendar error spacing:

Screen Shot 2021-03-12 at 11 30 49 AM

## Acceptance criteria
- [ ] Build passes

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/lib/moment-tz.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* istanbul ignore file */
```
Comment (jbalboni) [inline src/applications/vaos/lib/moment-tz.js:1]: Hiding this from our code coverage report

File: src/applications/vaos/project-cheetah/components/FormLayout.jsx [hunk 2]
```diff
@@ -9,12 +9,13 @@ import WarningNotification from '../../components/WarningNotification';
 
 export default function FormLayout({ children }) {
   const location = useLocation();
+  const match = useRouteMatch();
   return (
     <div className="vads-l-grid-container vads-u-padding-x--2p5 large-screen:vads-u-padding-x--0 vads-u-padding-bottom--8">
       <Breadcrumbs>
-        <Link to="new-covid-19-vaccine-booking">COVID-19 vaccine</Link>
+        <Link to={match.url}>COVID-19 vaccine</Link>
```
Comment (jbalboni) [inline src/applications/vaos/project-cheetah/components/FormLayout.jsx:16]: Switching to this means that the logic around being at the start of the flow works when the flow is set to the `/new-covid-19-vaccine-booking` (which is true when running the app normally) and also when it's set to `/` (which happens when we rendering just the `NewBooking` component in unit tests).

`match.url` is always the root url of the routes within the last `Switch` component (probably a better way of describing that).

File: src/applications/vaos/project-cheetah/components/SelectDate1Page.jsx [hunk 3]
```diff
@@ -185,6 +185,9 @@ export function SelectDate1Page({
               .add(395, 'days')
               .format('YYYY-MM-DD')}
             validationError={submitted ? validationError : null}
+            required
```
Comment (jbalboni) [inline src/applications/vaos/project-cheetah/components/SelectDate1Page.jsx:188]: This was not added after a recent refactoring ticket.

File: src/applications/vaos/project-cheetah/index.jsx [hunk 4]
```diff
@@ -39,20 +40,26 @@ export function NewBookingSection({
   const history = useHistory();
   const location = useLocation();
 
+  useEffect(() => {
+    openNewBookingPage(history);
+  }, []);
```
Comment (jbalboni) [inline src/applications/vaos/project-cheetah/index.jsx:45]: This only needs to run the check once, now that the action to fetch the data is separate from the check for when to redirect

File: src/applications/vaos/project-cheetah/index.jsx [hunk 5]
```diff
@@ -39,20 +40,26 @@ export function NewBookingSection({
...
     },
-    [isEligible],
+    [featureProjectCheetah, history],
   );
 
   useEffect(
     () => {
-      if (!featureProjectCheetah) {
-        history.push('/');
+      if (newBookingStatus === FETCH_STATUS.failed) {
```
Comment (jbalboni) [inline src/applications/vaos/project-cheetah/index.jsx:58]: Make sure we focus on the error message if the call fails

File: src/applications/vaos/tests/project-cheetah/components/NewBookingPage.unit.spec.jsx [hunk 6]
```diff
@@ -30,23 +44,18 @@ describe('VAOS <ConfirmedAppointmentDetailsPage>', () => {
...
-    // response.attributes.coreSettings[0].id = '301';
-    // response.attributes.coreSettings[0].typeOfCare = 'Vaccine';
-
     mockDirectBookingEligibilityCriteria(
       ['983', '984'],
       [
-        {
-          ...response,
-        },
+        getDirectBookingEligibilityCriteriaMock({
```
Comment (jbalboni) [inline src/applications/vaos/tests/project-cheetah/components/NewBookingPage.unit.spec.jsx:55]: Simplified the code for some of these tests

File: src/applications/vaos/tests/project-cheetah/components/ClinicChoicePage.unit.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+    await screen.findByText(/Choose a clinic located at/i);
+
+    expect(screen.baseElement).to.contain.text('Cheyenne VA Medical Center');
+    expect(screen.baseElement).to.contain.text('Cheyenne, WY 82001-5356');
+    expect(screen.baseElement).to.contain.text('307-778-7550');
+
+    expect(screen.getAllByRole('radio').length).to.equal(2);
+    expect(screen.getByLabelText('Green team clinic')).to.have.tagName('input');
+    expect(screen.getByLabelText('Red team clinic')).to.have.tagName('input');
```
Comment (vbahinwillit) [inline src/applications/vaos/tests/project-cheetah/components/ClinicChoicePage.unit.spec.jsx:92]: Is tagName more of an implementation detail?

Comment (jbalboni) [reply src/applications/vaos/tests/project-cheetah/components/ClinicChoicePage.unit.spec.jsx:92]: Yeah, should really search for these by role.

**Review by vbahinwillit — APPROVED** [review]
Just one question but LGTM.

**Pull request merged**